src/triggers.cxx

```diff
@@ -233,17 +233,19 @@ ROOT::RDF::RNode GenerateSingleTriggerFlag(
     // For v9 compatibility a type casting is applied
     auto [df1, triggerobject_id_column] = utility::Cast<ROOT::RVec<UShort_t>, ROOT::RVec<Int_t>>(
             df, triggerobject_id+"_v12", "ROOT::VecOps::RVec<UShort_t>", triggerobject_id);
-
+    
     auto triggermatch = [DeltaR_threshold, pt_cut, eta_cut,
                          trigger_particle_id_cut, triggerbit_cut, hltpath](
                             bool hltpath_match,
                             const ROOT::Math::PtEtaPhiMVector &particle_p4,
-                            ROOT::RVec<int> triggerobject_bits,
+                            ROOT::RVec<ULong64_t> triggerobject_bits,
                             ROOT::RVec<UShort_t> triggerobject_ids_v12,
                             ROOT::RVec<float> triggerobject_pts,
                             ROOT::RVec<float> triggerobject_etas,
                             ROOT::RVec<float> triggerobject_phis) {
         auto triggerobject_ids = static_cast<ROOT::RVec<int>>(triggerobject_ids_v12);
+        // In nanoAODv15 the type of trigger bits was changed to ULong64_t
+        auto triggerobject_bits_int = static_cast<ROOT::RVec<int>>(triggerobject_bits);
```

Comment:
You also need to add this `utility::Cast <ROOT::RVec<ULong64_t>, ROOT::RVec<Int_t>>(...)` line for the `triggerobject_bit`, otherwise this will not work for nanoAOD v9 and v12.